public/assets/reportcard/reportcard-80c0d560a7d7912c7d08c1d2f767fed7c3a3b32a22b54547ff47522588c211b6.js

```diff
@@ -0,0 +1 @@
),e.cacheError&&e.cacheError(o,r,a)}e.realsuccess&&e.realsuccess(t)},n&&"expired"!==n||localStorage.setItem(r+"cachettl",+new Date+36e5*s)))}),function(t){return t.fn.reportCard=function(o){var r,a,s,n,c,u;return u=function(e){return e.toString().replace(/\B(?=(\d{3})+(?!\d))/g,",")},a=function(){function e(e){this.userName=e.userName,this.site=e.site,this.badgesAmount=e.badgesAmount,this.tooltips=e.tooltips}return e.prototype.build=function(e){var r,a,s;a=e.badges.slice(-o.badgesAmount),s=100/(a.length+1)+"%",r="<h3>I have passed "+e.badge_count+" lessons and scored "+u(e.points_total)+" points at "+e.site+'!</h3>\n<p>Check out some of my last passed course content at the badges below: </p>\n<ul class="badges">',a.forEach(function(t){return r+='<li style="width: '+s+';" title="'+t.label+'">\n  <a href="'+e.profile_url+'" target="_blank" data-toggle="tooltip" data-placement="top" >\n    <img src="'+t.icon_url+'" alt="'+t.label+'"/>\n  </a>\n</li>'}),r+="</ul>",o.$element.html(r),o.tooltips&&o.$element.each(function(e,o){return t(o).find("li").tooltip()})},e.prototype.transform=function(e){return e},e.prototype.getData=function(e){var r;return r=this,t.ajax({type:"GET",url:this.url,localCache:!0,cacheTTL:.5,cacheKey:"reportcard"+o.site+o.userName,dataType:e.dataType,crossDomain:!0,async:!0,beforeSend:function(){o.$element.html('<div class="spinner"></div>')},success:function(e){return r.build(r.transform(e))}})},e}(),s=function(t){function o(e){o.__super__.constructor.call(this,e),this.url="https://teamtreehouse.com/"+this.userName+".json"}return e(o,t),o.prototype.getData=function(){return o.__super__.getData.call(this,{dataType:"json"})},o.prototype.transform=function(e){return{site:"Treehouse",username:e.profile_name,profile_url:e.profile_url,points:e.points,points_total:e.points.total,badge_count:e.badges.length,badges:e.badges.map(function(e){return{courses:e.courses,course_count:e.courses.length,earned_date:Date.parse(e.earned_date),icon_url:e.icon_url,label:e.name,url:e.url}})}},o}(a),r=function(t){function o(e){o.__super__.constructor.call(this,e),this.url="https://www.codeschool.com/users/"+this.userName+".json"}return e(o,t),o.prototype.getData=function(){return o.__super__.getData.call(this,{dataType:"jsonp"})},o.prototype.transform=function(e){return{site:"Code School",username:e.user.username,profile_url:"https://www.codeschool.com/users/"+e.user.username,points:void 0,points_total:e.user.total_score,badge_count:e.badges.length,badges:e.badges.map(function(e){return{courses:void 0,course_count:void 0,earned_date:void 0,icon_url:e.badge,label:e.name,url:e.course_url}})}},o}(a),o.$element=this,o.badgesAmount=o.badgesAmount?o.badgesAmount:5,o.userName||alert("You need to pass in a username"),o.site||alert("You need to pass in a site"),"treehouse"===o.site?(c=new s(o),c.getData()):"codeschool"===o.site?(n=new r(o),n.getData()):void 0}}(jQuery),$(".report-card.codeschool").reportCard({userName:"bibar",site:"codeschool",badgesAmount:15})});
```

Comment:
Line is too long.<br>Missing semicolon.<br>Expected '{' and instead saw 't'.<br>Expected an assignment or function call and instead saw an expression.<br>The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.<br>A constructor name should start with an uppercase letter.<br>Missing '()' invoking a constructor.<br>'e' is already defined.<br>Identifier 'badge_count' is not in camel case.<br>Identifier 'points_total' is not in camel case.<br>Mixed double and single quotes.<br>Identifier 'profile_url' is not in camel case.<br>Identifier 'icon_url' is not in camel case.<br>A leading decimal point can be confused with a dot: '.5'.<br>Wrap an immediate function invocation in parens to assist the reader in understanding that the expression is the result of a function, and not the function itself.<br>Too many errors. (100% scanned).